### ISSUE TITLE
Compile error in Manjaro last version

### ISSUE THREAD

**Issue by orderorder** [jonof/jftekwar#3]
```
src/bstub.c:1378:25: error: demasiados argumentos para la función ‘initengine’
 1378 |                 case 5: initengine(5,320L,200L); break;
      |                         ^~~~~~~~~~
En el fichero incluido desde src/bstub.c:1:
jfbuild/include/build.h:384:8: nota: se declara aquí
  384 | int    initengine(void);
      |        ^~~~~~~~~~
src/bstub.c:1379:25: error: demasiados argumentos para la función ‘initengine’
 1379 |                 case 6: initengine(6,320L,200L); break;
      |                         ^~~~~~~~~~
En el fichero incluido desde src/bstub.c:1:
jfbuild/include/build.h:384:8: nota: se declara aquí
  384 | int    initengine(void);
      |        ^~~~~~~~~~
src/bstub.c:1386:25: aviso: el puntero que apunta en el paso del argumento 2 de ‘makepalookup’ difiere en signo [-Wpointer-sign]
 1386 |         makepalookup(16,tempbuf,0,0,0,1);
      |                         ^~~~~~~
      |                         |
      |                         char *
En el fichero incluido desde src/bstub.c:1:
jfbuild/include/build.h:396:48: nota: se esperaba ‘unsigned char *’ pero el argumento es de tipo ‘char *’
  396 | int    makepalookup(int palnum, unsigned char *remapbuf, signed char r, signed char g, signed char b, unsigned char dastat);
      |                                 ~~~~~~~~~~~~~~~^~~~~~~~
src/bstub.c:1394:9: error: ‘kensplayerheight’ no se declaró aquí (primer uso en esta función)
 1394 |         kensplayerheight = 34;
      |         ^~~~~~~~~~~~~~~~
src/bstub.c:1395:9: error: ‘zmode’ no se declaró aquí (primer uso en esta función)
 1395 |         zmode = 0;
      |         ^~~~~
src/bstub.c:1396:9: error: ‘defaultspritecstat’ no se declaró aquí (primer uso en esta función); ¿quiso decir ‘nextspritestat’?
 1396 |         defaultspritecstat = 0;
      |         ^~~~~~~~~~~~~~~~~~
      |         nextspritestat
src/bstub.c: En la función ‘ExtUnInit’:
src/bstub.c:1414:9: aviso: declaración implícita de la función ‘uninitgroupfile’ [-Wimplicit-function-declaration]
 1414 |         uninitgroupfile();
      |         ^~~~~~~~~~~~~~~
src/bstub.c: En la función ‘ExtCheckKeys’:
src/bstub.c:1425:13: error: ‘qsetmode’ no se declaró aquí (primer uso en esta función); ¿quiso decir ‘qsetmodeany’?
 1425 |         if (qsetmode == 200)    //In 3D mode
      |             ^~~~~~~~
      |             qsetmodeany
src/bstub.c:1428:14: error: ‘somethingintab’ no se declaró aquí (primer uso en esta función)
 1428 |          if( somethingintab == 3 )   // prevent extra value copy
      |              ^~~~~~~~~~~~~~
src/bstub.c:1441:17: aviso: declaración implícita de la función ‘editinput’ [-Wimplicit-function-declaration]
 1441 |                 editinput();
      |                 ^~~~~~~~~
src/bstub.c:1446:14: error: ‘keystatus’ no se declaró aquí (primer uso en esta función)
 1446 |          if( keystatus[39] == 1 ) {  // ';' to clear all exts
      |              ^~~~~~~~~
src/bstub.c: En la función ‘ExtGetWallCaption’:
src/bstub.c:1576:10: error: ‘keystatus’ no se declaró aquí (primer uso en esta función)
 1576 |      if (keystatus[0x57] > 0) {    // f11   Grab pic 0x4e +
      |          ^~~~~~~~~
src/bstub.c:1578:49: aviso: formato ‘%ld’ espera un argumento de tipo ‘long int’, pero el argumento 3 es de tipo ‘int’ [-Wformat=]
 1578 |           sprintf(tempbuf,"Grabed Wall Picnum %ld",wallpicnum);
      |                                               ~~^  ~~~~~~~~~~
      |                                                 |  |
      |                                                 |  int
      |                                                 long int
      |                                               %d
src/bstub.c:1585:26: error: ‘posx’ no se declaró aquí (primer uso en esta función)
 1585 |                          posx=(wall[i].x)-(((wall[i].x)-(wall[wall[i].point2].x))/2);
      |                          ^~~~
src/bstub.c:1586:26: error: ‘posy’ no se declaró aquí (primer uso en esta función)
 1586 |                          posy=(wall[i].y)-(( (wall[i].y)-(wall[wall[i].point2].y))/2);
      |                          ^~~~
src/bstub.c:1600:26: error: ‘ang’ no se declaró aquí (primer uso en esta función)
 1600 |                          ang=sprite[i].ang;
      |                          ^~~
src/bstub.c: En la función ‘ExtShowWallData’:
src/bstub.c:1832:13: error: ‘qsetmode’ no se declaró aquí (primer uso en esta función); ¿quiso decir ‘qsetmodeany’?
 1832 |         if (qsetmode == 200)    //In 3D mode
      |             ^~~~~~~~
      |             qsetmodeany
src/bstub.c: En la función ‘ExtEditSectorData’:
src/bstub.c:1910:13: error: ‘qsetmode’ no se declaró aquí (primer uso en esta función); ¿quiso decir ‘qsetmodeany’?
 1910 |         if (qsetmode == 200)    //In 3D mode
      |             ^~~~~~~~
      |             qsetmodeany
src/bstub.c:1913:21: error: ‘searchstat’ no se declaró aquí (primer uso en esta función)
 1913 |                 if (searchstat == 1)
      |                     ^~~~~~~~~~
src/bstub.c:1914:32: error: ‘searchsector’ no se declaró aquí (primer uso en esta función)
 1914 |                         sector[searchsector].ceilingpicnum++;   //Just a stupid example
      |                                ^~~~~~~~~~~~
src/bstub.c:1922:44: aviso: formato ‘%ld’ espera un argumento de tipo ‘long int’, pero el argumento 3 es de tipo ‘int’ [-Wformat=]
 1922 |                 sprintf(tempbuf,"Sector (%ld) visibility: %d.",sectnum, visibility); //sector[sectnum].visibility); oog
      |                                          ~~^                   ~~~~~~~
      |                                            |                   |
      |                                            long int            int
      |                                          %d
src/bstub.c: En la función ‘ExtEditWallData’:
src/bstub.c:1936:13: error: ‘qsetmode’ no se declaró aquí (primer uso en esta función); ¿quiso decir ‘qsetmodeany’?
 1936 |         if (qsetmode == 200)    //In 3D mode
      |             ^~~~~~~~
      |             qsetmodeany
src/bstub.c:1941:42: aviso: formato ‘%ld’ espera un argumento de tipo ‘long int’, pero el argumento 3 es de tipo ‘int’ [-Wformat=]
 1941 |                 sprintf(tempbuf,"Wall (%ld) Nick's variable: ",wallnum);
      |                                        ~~^                     ~~~~~~~
      |                                          |                     |
      |                                          long int              int
      |                                        %d
src/bstub.c: En la función ‘ExtEditSpriteData’:
src/bstub.c:1976:11: error: ‘keystatus’ no se declaró aquí (primer uso en esta función)
 1976 |           keystatus[0x42]=0;
      |           ^~~~~~~~~
src/bstub.c:1985:40: aviso: formato ‘%ld’ espera un argumento de tipo ‘long int’, pero el argumento 3 es de tipo ‘int’ [-Wformat=]
 1985 |      sprintf(tempbuf,"Current Sprite %ld",cursprite);
      |                                      ~~^  ~~~~~~~~~
      |                                        |  |
      |                                        |  int
      |                                        long int
      |                                      %d
src/bstub.c: En el nivel principal:
src/bstub.c:1992:1: aviso: el tipo de devolución por defecto es ‘int’ [-Wimplicit-int]
 1992 | faketimerhandler()
      | ^~~~~~~~~~~~~~~~
src/bstub.c:1992:1: aviso: tipos en conflicto para ‘faketimerhandler’
En el fichero incluido desde src/bstub.c:1:
jfbuild/include/build.h:262:13: nota: previous declaration of ‘faketimerhandler’ with type ‘void(void)’
  262 | extern void faketimerhandler(void);
      |             ^~~~~~~~~~~~~~~~
make: *** [Makefile:202: src/bstub.o] Error 1

```

**Comment by jonof** [comment]
That's expected at the moment. The game will compile but I haven't updated the editor stub to make it work yet. It may be a Christmas holiday task.

**Comment by orderorder** [comment]
ok. I love all these projects that bring old titles to nowdays. Thanks for your effort

**Comment by Exarkuniv** [comment]
is this still being worked on anymore?

**Comment by jonof** [comment]
Occasionally, usually when I'm bored over Christmas periods.

**Comment by jonof** [comment]
Ought be solved in 9d9588a
